instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -203,9 +204,13 @@ async def __call__(self, scope, receive, send):
         token = context.attach(extract(scope, getter=asgi_getter))
         span_name, additional_attributes = self.span_details_callback(scope)
 
+        span_kind = trace.SpanKind.INTERNAL
+        current_span = trace.get_current_span()
+        if current_span is None or not isinstance(current_span, Span):
+            span_kind = trace.SpanKind.SERVER
```

Comment:
Why not use `ServerSpanTracer` here as well?

Reply:
I didn't use `ServerSpanTracer` here because for the asgi instrumentation, the span is created within a `try` block, which `ServerSpanTracer` does not implement because all the other instrumentations don't use one.